### PR TITLE
fix link navigation to extended config file in tsconfig

### PR DESCRIPTION
**Pull request by vitaliymaz**

### CODE REVIEW

**Comment by vitaliymaz** [comment]
Fixes #71639

File: extensions/typescript-language-features/src/features/tsconfig.ts [hunk 1]
```diff
@@ -26,23 +26,30 @@ class TsconfigLinkProvider implements vscode.DocumentLinkProvider {
...
 	}
 
-	private getExendsLink(document: vscode.TextDocument, root: jsonc.Node): vscode.DocumentLink | undefined {
+	private getExtendsLink(document: vscode.TextDocument, root: jsonc.Node): vscode.DocumentLink | undefined {
 		const extendsNode = jsonc.findNodeAtLocation(root, ['extends']);
 		if (!this.isPathValue(extendsNode)) {
 			return undefined;
 		}
 
+		if (extendsNode.value.startsWith('./') || extendsNode.value.startsWith('../')) {
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/features/tsconfig.ts]: Not sure if TS supports windows style paths but to be safe I think this condition can just be `startsWith('.')`.

File: extensions/typescript-language-features/src/features/tsconfig.ts [hunk 2]
```diff
@@ -26,23 +26,30 @@ class TsconfigLinkProvider implements vscode.DocumentLinkProvider {
...
+			return new vscode.DocumentLink(
+				this.getRange(document, extendsNode),
+				vscode.Uri.file(join(dirname(document.uri.fsPath), extendsNode!.value + (basename(extendsNode.value).match('.json$') ? '' : '.json')))
```
Comment (mjbvz) [inline extensions/typescript-language-features/src/features/tsconfig.ts]: Use `endsWith` instead of `match` (so we don't have to escape the `.`)

**Review by mjbvz — CHANGES_REQUESTED** [review]
Thanks for taking a look. Just a few minor changes and I think we can merge this for April

**Comment by mjbvz** [comment]
Thanks! Change will be in the next VS Code insiders build and is scheduled to ship with VS code 1.34

**Pull request merged**